File: src/main/java/br/com/blz/testjava/interfaces/IProductService.java
package br.com.blz.testjava.interfaces;

import br.com.blz.testjava.entity.Product;

public interface IProductService {
	public void create(Product product);
	
	public Boolean isExists(Long sku);
	
	public void delete(Long sku);

	public Product reader(Long sku);
}

File: src/main/java/br/com/blz/testjava/product/service/IProductService.kt
package br.com.blz.testjava.product.service

import br.com.blz.testjava.product.dto.ProductInDTO
import br.com.blz.testjava.product.dto.ProductOutDTO

interface IProductService {
  fun create(productInDTO: ProductInDTO): ProductOutDTO
  fun editBySku(sku: Long, productInDTO: ProductInDTO): ProductOutDTO
  fun findBySku(sku: Long): ProductOutDTO
  fun deleteBySku(sku: Long)
}
